controllers/datafoundationclient.go

```diff
@@ -80,7 +80,7 @@ func dfcGetOfferingSpecInstance() *dataFoundationClientSpec {
 func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering, offeringSpec interface{}) (ctrl.Result, error) {
 	r := dataFoundationClientReconciler{}
 	r.initReconciler(offeringReconciler, offering)
-	r.spec = offeringSpec.(dataFoundationClientSpec)
+	r.spec = *offeringSpec.(*dataFoundationClientSpec)
```

Comment:
I believe this is wrong. you are creating a copy of the `dataFoundationClientSpec` and then a copy of the `offeringSpec`. I believe the issue here is that `r.spec` is not a pointer (ref type)